kernel/base/src/main/java/com/twosigma/beakerx/kernel/commands/MagicCommand.java

```diff
@@ -144,11 +148,30 @@ public MagicCommandFunctionality addImport() {
       if (parts.length != 2) {
         return createResultWithCustomMessage(kernel.getImports().toString(), message, executionCount);
       }
-      this.kernel.addImport(new ImportPath(parts[1]));
-      return getMagicCommandItem(code, message, executionCount);
+
+      if (isValidImport(parts[1], executionCount)) {
+        this.kernel.addImport(new ImportPath(parts[1]));
+        return getMagicCommandItem(code, message, executionCount);
+      }
+
+      return sendErrorMessage(message, "Could not import " + parts[1] + ", class not found.",
+          executionCount);
     };
   }
 
+  private boolean isValidImport(String part, int executionCount) {
+    try {
+      part = part.replace(";","");
```

Comment:
i don't understand what this is for, there shouldn't be a semicolon, if there is it should be a syntax error reported to the user.